File: Programs/Move_Zeroes.py
""" LeetCode """
# Append & Remove: O(n^2)
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        for num in nums:
            if num == 0:
                nums.append(num)
                nums.remove(num)
# Two Pointers: O(n)
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        i= 0
        for j in range(len(nums)):
            if nums[j] != 0:
                nums[i], nums[j]= nums[j], nums[i]
                i += 1

""" CodingNinjas """
def pushZerosAtEnd(arr):
    i= 0
    for j in range(len(arr)):
        if arr[j] != 0:
            arr[i], arr[j]= arr[j], arr[i]
            i += 1


""" GeeksForGeeks """
class Solution:
    def pushZerosToEnd(self, arr):
        i= 0
        for j in range(len(arr)):
            if arr[j] != 0:
                arr[i], arr[j] = arr[j], arr[i]
                i += 1

File: Programs/Move_Zeroes.cpp
#include<bits/stdc++.h>
using namespace std;

/* LeetCode */
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n= nums.size();
        
        for(int i=0, j=0; j<n; j++) {
            if(nums[j] != 0) {
                swap(nums[i], nums[j]);
                i++;
            }
        }
    }
};

/* CodingNinjas */
void pushZerosAtEnd(vector<int> &arr) {
    for(int i=0, j=0; j<arr.size(); j++) {
        if(arr[j] != 0) {
            swap(arr[i], arr[j]);
            i++;
        }
    }
}

/* GeeksForGeeks */
class Solution {
  public:
    void pushZerosToEnd(vector<int>& arr) {
        int n= arr.size();
        
        for(int i=0, j=0; j<n; j++) {
            if(arr[j] != 0) {
                swap(arr[i], arr[j]);
                i++;
            }
        }
    }
};